web/src/pages/TracesPage/TracesPage.tsx

```diff
@@ -32,19 +29,12 @@ export function TracesPage() {
             <Title headingLevel="h1">{t('Traces')}</Title>
             <Divider />
             <Split hasGutter>
-              <TempoStackDropdown
-                id="tempostack-dropdown"
-                tempoStackOptions={tempoStackList}
-                selectedTempoStackName={tempoStack}
-                selectedNamespace={namespace}
-                setTempoList={setTempoStackInURL}
-                isLoading={loading}
-              />
-              <DurationDropdown handleDurationChange={setDuration} />
+              <TempoInstanceDropdown />
+              <DurationDropdown duration={duration as DurationString} setDuration={setDuration} />
```

Comment:
instead of casting, should we validate the query param here and define some defaults in case of failure or absence?

Reply:
Good point. I switched to the enum type in use-query-params, so now if the duration is invalid, the default duration is used instead of an invalid one.